PROVIDERS.rst

```diff
@@ -0,0 +1,243 @@
+as they see fit, as long as it does not cause conflicting dependencies.
+
+You can read more about it in the
+`Installation and upgrade scenarios <https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#installation-and-upgrade-scenarios>`_
+chapter of our user documentation.
+
+Community managed providers
+===========================
+
+When providers are accepted by the community, the providers must follow the Apache Software Foundation
+rules and policies when it comes to accepting contributions and releasing new versions of the providers. This
+means that the code changes in the providers must be reviewed by Airflow committers and merged when they
+are accepted by them, also they must implement all the test and documentation requirements that are
+required by the community.
+
+They providers in their latest version in "main" branch of airflow repository, are also tested together
+with other community providers and one of the key properties of the community providers is that the latest
+version of providers contribute their dependencies to constraints of  Airflow published when Airflow Core is
+released - which means that when you are using constraints published  by Airflow, you can install all
+the providers together and they are tested together. This allows to add an optional "extra" to Airflow
+for each provider, so that the providers can be installed together with Airflow by specifying the
+"extra" in the installation command.
+
+Because of the constraint and potential conflicting dependencies, the community providers have to be regularly
+updated and the community might decide to suspend releases of a provider if we find out that we have trouble
+with updating the dependencies of the provider, or if we find out that the provider is not compatible with
+other providers. See the section below for more details on suspending releases of the community providers.
+
+List of all available community providers is available at the `Providers index <https://airflow.apache.org/docs/>`_.
+
+Accepting new community providers
+=================================
+
+Accepting new community providers should be a deliberate process that requires ``[DISCUSSION]``
+followed by ``[VOTE]`` thread at the airflow `devlist <https://airflow.apache.org/community/#mailing-list>`_.
+In case of provider is integrating with an open-source software rather than service (particularly if the
+open-source software is an Apache Software Foundation, Linux Software Foundation or similar organisation with
+well established governance processes that are not strictly vendor-controlled) and when the software is
+well established an popular it might be enough to have a good and complete PR of the provider, ideally
+with a good test coverage (including integration tests) and documentation approved by the team
+``[LAZY CONSENSUS]`` on the devlist might be enough to accept the provider.
```

Comment:
I think we have a loop hole here.
If I take for example the google case. The current state allows to add any new google integration without passing trough this procedure because everything is packed under 1 provider.

I think we should also specify handling case of adding new SDK under existed provider that extend the provider to a new service that wasn't part of it to begin with. If we take for example when we first added Tableau to the current existed provider of Salesforce following the current policy states that there is no need for a mailing list vote on it which seems wrong?

Reply:
I am not sure we have to go that deep. I am more concerned about  requests to add a new provider/integration thn a new "dependency" to an existing provider. We already get "historically" some providers that we manage, and that's sometihing that is both - the strength of Airflow and weakeness (dependency). And in general, any committer might block merging a new dependency being added when we think it is a problem. We do not have specifically explain that as a process (I think). Or when we see that  a new operator does not fit.  And we can always start a discussion.

The process for adding a new provider is slightly different, I think - this is mostly about new integration, potentially new team of people from a service interacting with us, and a huge risk that they want to "drop" maintenance on us and won't be contributing new things, hoping that people in the community will and they can just walk away. 

I think when getting new PR in an existing provider we have, we have much better knowledge - will there be maintenance, will there be support in case of problem, what happens when there is a need to upgrade or when the service gets disabled. We are going through  that with Google team and while a bit late, the make a huge effort to make their provider updated now. This is a good lesson for us.

So I  think while we **could** describe it in more detail, I tihnk this is really more of an explanation for anyone "new" (like Huawei Cloud service people) players that need to understand what it means to add a new provider. 

Does it make sense?

Reply:
Also this is more a "default" behaviour. At any point, anyone can call for VOTE, LAZY CONSENSUS on just about anything, this is more like "what to expect" than "strict rules".

Reply:
> I think when getting new PR in an existing provider we have, we have much better knowledge - will there be maintenance, will there be support in case of problem, what happens when there is a need to upgrade or when the service gets disabled. We are going through that with Google team and while a bit late, the make a huge effort to make their provider updated now. This is a good lesson for us.

I'm not referring to the case that we have now.
I'm referring for a case that adding a new extension for a service that is not part of the original provider scope. If tomorrow Google acquire a company technically it allows them to add a new integration for it without violating this policy. I don't mean to cases where under google.cloud we add more services and improve the support there i'm referring to cases of adding a whole new service branch like: leveldb, looker etc...